app/views/contexts/section_summary_context.py

```diff
@@ -65,16 +65,25 @@ def _build_summary(self, location):
 
     def _title_for_location(self, location):
 
-        title = self._schema.get_section(location.section_id).get("title")
-        if location.list_item_id:
-            repeating_title = self._schema.get_repeating_title_for_section(
-                location.section_id
-            )
+        section_id = location.section_id
+        list_item_id = location.list_item_id
+
+        if list_item_id:
+            repeating_title = self._schema.get_repeating_title_for_section(section_id)
             if repeating_title:
-                title = self._placeholder_renderer.render_placeholder(
-                    repeating_title, location.list_item_id
+                return self._placeholder_renderer.render_placeholder(
```

Comment:
This is a pain, and why I can't get this block a bit neater. For section summary titles I must check for a placeholder as it might not have one, however repeating is assuming one.

Although completely logical to assume a placeholder for repeating, how validator is currently written it is possible to have a valid schema without a placeholder in repeating meaning this will blow up.

Reply:
I find it confusing that we have multiple methods to get a section title, we have `get_repeating_title_for_section`, `get_summary_title_for_section` and `get_title_for_section`.
I think we should have a single method that return the title based on the priority, so:
```python
    def get_title_for_section(self, section_id):
        return (
            self.get_repeating_title_for_section(section_id)
            or self.get_summary_title_for_section(section_id)
            or self._sections_by_id.get(section_id).get("title")
        )
```

Doing this would mean, we can simplify the method above to:
```python
    def _title_for_location(self, location):
        title = self._schema.get_title_for_section(location.section_id)
        if isinstance(title, dict):
            return self._placeholder_renderer.render_placeholder(
                title, location.list_item_id
            )

        return title
```
You could additionally add `placeholders in `title` but at the moment, title can either be a string or a dict, and a dict implies a placeholder.

Reply:
Since the the hub requires the default section title, you would need to call it directly.
```python
    def _title_for_location(self, location):
        section_id = location.section_id
        title = (
            self._schema.get_repeating_title_for_section(section_id)
            or self._schema.get_summary_title_for_section(section_id)
            or self._schema.get_title_for_section(section_id)
        )
        if isinstance(title, dict):
            return self._placeholder_renderer.render_placeholder(
                title, location.list_item_id
            )

        return title

Reply:
You might have used more characters than the PR on this comment, but I like the dict suggestion.